homeassistant/components/google/__init__.py

```diff
@@ -116,7 +115,6 @@ def scope(self) -> str:
         vol.Optional(CONF_OFFSET): cv.string,
         vol.Optional(CONF_SEARCH): cv.string,
         vol.Optional(CONF_TRACK): cv.boolean,
-        vol.Optional(CONF_MAX_RESULTS): cv.positive_int,
```

Comment:
Probably mark this key as deprecated to avoid an immediate breaking change for users that have this key in their config. We can use `cv.deprecated`. Keep the key in the schema here for now and use `cv.deprecated` together with `vol.All` around the dict schema.

Reply:
With the new code, this config is obsolete. All appointments will be fetched. Nothing breaks: users that still have the ``max_results`` setting won't have trouble.

Reply:
If we remove the key from the validation schema, config validation will fail.

Reply:
I'm sorry, I didn't know that. It didn't fail in my testing. Then I'll bring it back as deprecated.

Reply:
See the update